CRM/Contribute/BAO/Contribution.php

```diff
@@ -436,6 +420,22 @@ public static function create(&$params, $ids = array()) {
 
     $transaction->commit();
 
```

Comment:
Just to be safe, how about adding check to ensure contribution object exist. ( just incase of rollback )